src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1934,21 +1935,27 @@ File createTempFile(String prefix, String suffix) throws IOException {
         }
         Set<PosixFilePermission> ownerOnly = PosixFilePermissions.fromString("rw-------");
         FileAttribute<Set<PosixFilePermission>> fileAttribute = PosixFilePermissions.asFileAttribute(ownerOnly);
-        return Files.createTempFile(tmpPath, prefix, suffix, fileAttribute).toFile();
+        return Files.createTempFile(tmpPath, prefix, suffix, fileAttribute);
     }
 
-    private void deleteTempFile(File tempFile) {
-        if (tempFile != null && !tempFile.delete() && tempFile.exists()) {
-            listener.getLogger().println("[WARNING] temp file " + tempFile + " not deleted");
+    private void deleteTempFile(Path tempFile) {
+        if (tempFile != null) {
+            try {
+                Files.deleteIfExists(tempFile);
+            } catch (IOException e) {
+                Functions.printStackTrace(e, listener.error("temp file " + tempFile + " not deleted"));
```

Comment:
I accepted the suggestion, but on reading the code I cannot think of any legitimate reason why we would be able to create the temporary file but not delete it. We're always closing the file, so there should never be an open file handle (which would prevent us from deleting it on Windows). If we had the permissions to create it, we should have the permissions to delete it. So to me this is a truly exceptional exception (the best kind!), as it indicates either a programming error (e.g., we forgot to close the file handle somewhere and we're on WIndows so the deletion fails) or a pathological runtime error (e.g., the OS is failing deletion requests due for some bizarre reason - copy-on-write filesystems like ZFS for example can fail deletions when the filesystem is full because a "deletion" requires adding some new metadata to indirect blocks). In either case, I would want the stack trace to be preserved _somewhere_ for analysis and losing it seems tragic. But I could see the argument for not burdening the user with this in the build console output. How would you feel about logging this at warning level to the Jenkins log?

Reply:
> But I could see the argument for not burdening the user with this in the build console output. How would you feel about logging this at warning level to the Jenkins log?

I think that is a great idea to log it at warning level in the Jenkins log.

Reply:
I replaced the build console log warning with a Jenkins log warning including the stack trace. I didn't keep the warning in the build console log because I felt it was not actionable for the end user and that logging things twice is pointless, but I don't feel strongly about that either.